### PR TITLE
Fixes #2528 - Getting contains no value for name href error from back…

### LINKED ISSUE #2528: Getting contains no value for name `href` error from backend for teams PATCH API.
**Affected module**
Backend and UI

**Describe the bug**
I was trying to remove the user from the team users list, but 
Getting `contains no value for name 'href` error from backend for teams PATCH API.

**To Reproduce**
Step 1: Take any team object
Step2 : Create Patch from [here](https://extendsclass.com/json-patch.html)
Step3: Now make a request to teams PATCH API (http://localhost:8585/api/v1/teams/{id}) with patch payload.

**Expected behaviour**
It should have updated the team users list with proper values.

**Version:**
 - Python version:
 - OpenMetadata version: [e.g. 0.8]
 - OpenMetadata Ingestion package version: [e.g. `openmetadata-ingestion[docker]==XYZ`]

**Additional context**

### PR DESCRIPTION
**Pull request by sureshms**
…end for teams PATCH API

### Describe your changes :

See #2528 

#
### Type of change :

- [x] Bug fix
- [] Improvement
- [] New feature
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/teams/TeamResourceTest.java [hunk 1]
```diff
@@ -296,6 +324,16 @@ public void compareEntities(Team expected, Team updated, Map<String, String> aut
 
   @Override
   public void assertFieldChange(String fieldName, Object expected, Object actual) throws IOException {
-    assertCommonFieldChange(fieldName, expected, actual);
+    if (expected == actual) {
```
Comment (mithmatt) [inline catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/teams/TeamResourceTest.java:327]: Should this be `expected.equals(actual)` to compare the values of objects?

File: catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/teams/TeamResourceTest.java [hunk 2]
```diff
@@ -154,6 +158,30 @@ void patch_teamAttributes_as_non_admin_403(TestInfo test) throws HttpResponseExc
     assertResponse(exception, FORBIDDEN, "Principal: CatalogPrincipal{name='test'} is not admin");
   }
 
+  @Test
+  void patch_deleteUserFromTeam_200(TestInfo test) throws IOException {
+    // Add team to user relationships while creating a team
```
Comment (vivekratnavel) [inline catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/teams/TeamResourceTest.java:163]: Nit: update the comment

**Comment by vivekratnavel** [comment]
Can you also take care of the unit test failures?
```
2022-02-08T01:34:52.3244780Z [ERROR] Failures: 
2022-02-08T01:34:52.3246245Z [ERROR]   WebhookResourceTest.put_updateEndpointURL:178->EntityResourceTest.deleteEntity:1326->EntityResourceTest.deleteEntity:1336 Expected org.apache.http.client.HttpResponseException to be thrown, but nothing was thrown.
2022-02-08T01:34:52.3247085Z [ERROR]   RoleResourceTest>EntityResourceTest.afterAllTests:384 expected: <96> but was: <97>
```

**Comment by sureshms** [comment]
> Can you also take care of the unit test failures?
> 
> ```
> 2022-02-08T01:34:52.3244780Z [ERROR] Failures: 
> 2022-02-08T01:34:52.3246245Z [ERROR]   WebhookResourceTest.put_updateEndpointURL:178->EntityResourceTest.deleteEntity:1326->EntityResourceTest.deleteEntity:1336 Expected org.apache.http.client.HttpResponseException to be thrown, but nothing was thrown.
> 2022-02-08T01:34:52.3247085Z [ERROR]   RoleResourceTest>EntityResourceTest.afterAllTests:384 expected: <96> but was: <97>
> ```

These intermittent failures are unrelated to this patch. Let's create a separate issue.

File: catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/teams/TeamResourceTest.java (hunk 1 above)
Comment (sureshms) [reply catalog-rest-service/src/test/java/org/openmetadata/catalog/resources/teams/TeamResourceTest.java:327]: This is actually == check. This catches both objects being null...

**Review by vivekratnavel — APPROVED** [review]

**Pull request merged**